### PR TITLE
Add user application records creation

### PR DESCRIPTION
**Pull request by AN-BC**
Related to #666

@JS-BC - please review - did this with Copilot Workspaces as  a fun demo

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/MemberJunction/MJ/pull/668?shareId=8d525dff-f27a-4bf4-925c-5a81237e3109).

### CODE REVIEW

**Comment by JS-BC** [comment]
@AN-BC code generated was close, I made some adjustments

File: mj.config.cjs [hunk 1]
```diff
@@ -201,10 +201,12 @@ const mjServerConfig = {
     autoCreateNewUsers: true,
     newUserLimitedToAuthorizedDomains: false,
     newUserAuthorizedDomains: [],
-    newUserRoles: ['UI'],
+    newUserRoles: ['UI', 'Developer'],
```
Comment (AN-BC) [inline mj.config.cjs:204]: @JS-BC  i still don't like auto-adding people to the Developer role. It makes me nervous for this to be the default. Remind me, why are we doing that? Maybe just make `UI` slightly more capable instead?

Comment (JS-BC) [reply mj.config.cjs:204]: If we modify the UI role to have more capabilities, then I think it should work like how the Basic role is defined in this issue: https://github.com/MemberJunction/MJ/issues/671

**Comment by AN-BC** [comment]
Ok. I’m fine with the developer role being default then, I don’t want to do the other issue for now, let’s close that one out. We need a broader think on users/roles/settings in the system for authorization as we look ahead to version 3.0 of the framework. Let’s make sure our installation docs explain the config file and that this is the default role setup.

**Pull request merged**